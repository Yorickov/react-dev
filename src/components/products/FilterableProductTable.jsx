import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SearchBar from './SearchBar.jsx';
import ProductTable from './ProductTable.jsx';

class FilterableProductTable extends Component {
  state = { filterText: '', inStockOnly: false };

  handleFilterTextChange = ({ target: { value } }) => {
    this.setState({ filterText: value });
  };

  handleInStockChange = ({ target: { checked } }) => {
    this.setState({ inStockOnly: checked });
  };

  render() {
    const { products } = this.props;
    const { filterText, inStockOnly } = this.state;

    return (
      <div>
        <SearchBar
          filterText={filterText}
          inStockOnly={inStockOnly}
          onFilterTextChange={this.handleFilterTextChange}
          onInStockChange={this.handleInStockChange}
        />
        <ProductTable
          products={products}
          filterText={filterText}
          inStockOnly={inStockOnly}
        />
      </div>
    );
  }
}

FilterableProductTable.propTypes = {
  products: PropTypes.array,
};

export default FilterableProductTable;
