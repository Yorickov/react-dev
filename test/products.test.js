import React from 'react';
import FilterableProductTable from '../src/components/products/FilterableProductTable.jsx';

const products = [
  { category: 'Sporting Goods', price: '$49.99', stocked: true, name: 'Football' },
  { category: 'Sporting Goods', price: '$29.99', stocked: false, name: 'Basketball' },
  { category: 'Electronics', price: '$99.99', stocked: true, name: 'iPod Touch' },
  { category: 'Electronics', price: '$399.99', stocked: false, name: 'iPhone 5' },
];

it('FilterableProductTable', () => {
  const wrapper = mount(<FilterableProductTable products={products} />);
  expect(wrapper.render()).toMatchSnapshot();

  const filter = wrapper.find('[data-test="filter"]');
  const checked = wrapper.find('[data-test="checked"]');

  filter.simulate('change', { target: { value: 'ball' } });
  expect(wrapper.render()).toMatchSnapshot();

  filter.simulate('change', { target: { value: '' } });
  checked.simulate('change', { target: { checked: true } });
  expect(wrapper.render()).toMatchSnapshot();
});
