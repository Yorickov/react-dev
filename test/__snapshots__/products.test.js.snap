// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterableProductTable 1`] = `
<div>
  <form>
    <div
      class="mb-3"
    >
      <input
        class="form-control"
        data-test="filter"
        placeholder="Search..."
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <input
        class="form-check"
        data-test="checked"
        type="checkbox"
      />
    </div>
    <p>
       Only show products in stock
    </p>
  </form>
  <table
    class="table table-bordered"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Price
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          colspan="2"
        >
          Sporting Goods
        </th>
      </tr>
      <tr>
        <td>
          Football
        </td>
        <td>
          $49.99
        </td>
      </tr>
      <tr>
        <td>
          <span
            style="color: red;"
          >
            Basketball
          </span>
        </td>
        <td>
          $29.99
        </td>
      </tr>
      <tr>
        <th
          colspan="2"
        >
          Electronics
        </th>
      </tr>
      <tr>
        <td>
          iPod Touch
        </td>
        <td>
          $99.99
        </td>
      </tr>
      <tr>
        <td>
          <span
            style="color: red;"
          >
            iPhone 5
          </span>
        </td>
        <td>
          $399.99
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`FilterableProductTable 2`] = `
<div>
  <form>
    <div
      class="mb-3"
    >
      <input
        class="form-control"
        data-test="filter"
        placeholder="Search..."
        type="text"
        value="ball"
      />
    </div>
    <div
      class="mb-3"
    >
      <input
        class="form-check"
        data-test="checked"
        type="checkbox"
      />
    </div>
    <p>
       Only show products in stock
    </p>
  </form>
  <table
    class="table table-bordered"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Price
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          colspan="2"
        >
          Sporting Goods
        </th>
      </tr>
      <tr>
        <td>
          Football
        </td>
        <td>
          $49.99
        </td>
      </tr>
      <tr>
        <td>
          <span
            style="color: red;"
          >
            Basketball
          </span>
        </td>
        <td>
          $29.99
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`FilterableProductTable 3`] = `
<div>
  <form>
    <div
      class="mb-3"
    >
      <input
        class="form-control"
        data-test="filter"
        placeholder="Search..."
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <input
        class="form-check"
        data-test="checked"
        type="checkbox"
      />
    </div>
    <p>
       Only show products in stock
    </p>
  </form>
  <table
    class="table table-bordered"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Price
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          colspan="2"
        >
          Sporting Goods
        </th>
      </tr>
      <tr>
        <td>
          Football
        </td>
        <td>
          $49.99
        </td>
      </tr>
      <tr>
        <th
          colspan="2"
        >
          Electronics
        </th>
      </tr>
      <tr>
        <td>
          iPod Touch
        </td>
        <td>
          $99.99
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
